src/shipit_static_analysis/shipit_static_analysis/workflow.py

```diff
@@ -128,19 +128,25 @@ def run(self, revision):
             # Analyze files in revision
             revision.analyze_files(self.hg)
 
-        # Only run mach if revision has any C/C++ files
-        if revision.has_clang_files:
-            with stats.api.timer('runtime.mach'):
-                # mach configure with mozconfig
-                logger.info('Mach configure...')
-                run_check(['gecko-env', './mach', 'configure'], cwd=self.repo_dir)
-
-                # Build CompileDB backend
-                logger.info('Mach build backend...')
-                cmd = ['gecko-env', './mach', 'build-backend', '--backend=CompileDB']
-                run_check(cmd, cwd=self.repo_dir)
-        else:
-            logger.info('No clang files detected, skipping mach')
+        with stats.api.timer('runtime.mach'):
+            # Only run mach if revision has any C/C++ files
+            if revision.has_clang_files:
+                    # mach configure with mozconfig
+                    logger.info('Mach configure...')
+                    run_check(['gecko-env', './mach', 'configure'], cwd=self.repo_dir)
+
+                    # Build CompileDB backend
+                    logger.info('Mach build backend...')
+                    cmd = ['gecko-env', './mach', 'build-backend', '--backend=CompileDB']
+                    run_check(cmd, cwd=self.repo_dir)
+
+            else:
+                logger.info('No clang files detected, skipping mach')
+
+            # Setup python environment
```

Comment:
We probably don't need to do this in the `has_clang_files` case since it's done automatically the first time you use mach, but since it's so cheap it doesn't really matter.